timechart/utilities/utils.py

```diff
@@ -4,21 +4,45 @@
 from qtpy.QtGui import QColor
 from qtpy.QtWidgets import QMessageBox
 
+
+global STRIPTOOL_PREDEFINED_COLORS
+STRIPTOOL_PREDEFINED_COLORS = []
+
+
 PREDEFINED_COLORS = (QColor(Qt.red), QColor(Qt.green),  QColor(Qt.darkRed),  QColor(Qt.blue), QColor(Qt.darkGreen),
                      QColor(Qt.cyan), QColor(Qt.darkCyan), QColor(Qt.magenta), QColor(Qt.darkMagenta),
                      QColor(255, 0, 127), QColor(0, 85, 255), QColor(255, 85, 0), QColor(0, 170, 127),
                      QColor(0, 170, 255), QColor(85, 170, 255), QColor(255, 170, 0), QColor(255, 170, 255))
 
 
-def random_color():
+def add_striptool_color(color):
+    global STRIPTOOL_PREDEFINED_COLORS
+    STRIPTOOL_PREDEFINED_COLORS.append(color)
+
+
+def random_color(curve_colors_only=False):
     """
     Return a QColor object from a predefined list.
 
+    Parameters
+    ----------
+    curve_colors_only : bool
+        If True, exclude black or white as the randomized color since a curve could be blended into the background. If
+        False, include black or white as one of the randomized colors.
+
     Returns
     -------
     A color object randomly selected from the predefined list.
     """
-    return random.choice(PREDEFINED_COLORS)
+    while True:
```

Comment:
We want to keep picking a new color from a pool if the color is black or white or None (unlikely) if the curve_colors_only flag is set. If it's not set, we'll just take a color pick and break.

The intention here is to avoid picking black or white for a curve's color because that would make the curve blend into the chart's background. Maybe checking for None is unnecessary, but the while loop construct is to ensure the above criteria.

Reply:
I think Teddy is saying that there is no real purpose for this being a `while True` loop. This could just be a simple direct `random.choice` between all available colors, except for the ones that cause issues.

In general you should avoid a `while True` construct whenever it isn't 100% necessary to save yourself some headache in the future when you go back to add/change something to this code block. It's really easy to accidentally get into an infinite loop, and you're setting a trap for future you or some other unsuspecting programmer.

Reply:
Thanks, @ZLLentz and @teddyrendahl . I just wanted to make the code a little shorter. I removed the while True loop and replaced it with the equivalent code.